packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/chat.tests.js

```diff
@@ -0,0 +1,343 @@
+
+                const text = 'new text message';
+
+                keyboardMock(this.$input)
+                    .focus()
+                    .type(text);
+
+                this.$sendButton.trigger('dxclick');
+
+                assert.strictEqual(onMessageSend.callCount, 1);
+            });
+
+            QUnit.test('new message should be created after clicking the send button', function(assert) {
+                const text = 'new text message';
+
+                keyboardMock(this.$input)
+                    .focus()
+                    .type(text);
+
+                this.$sendButton.trigger('dxclick');
+
+                const $bubbles = this.$element.find(`.${CHAT_MESSAGE_BUBBLE_CLASS}`);
+                const bubble = $bubbles[$bubbles.length - 1];
+
+                assert.strictEqual($(bubble).text(), text);
+            });
+
+            QUnit.test('New message should be correct after clicking the send button', function(assert) {
+                assert.expect(3);
+
+                const text = 'new text message';
+
+                this.instance.option({
+                    onMessageSend: ({ message }) => {
+                        const { author, text: messageText } = message;
+
+                        assert.strictEqual(author, this.instance.option('user'), 'author field is correct');
+                        // eslint-disable-next-line no-prototype-builtins
+                        assert.strictEqual(message.hasOwnProperty('timestamp'), true, 'timestamp field is set');
+                        assert.strictEqual(messageText, text, 'text field is correct');
+                    },
+                });
+
+                keyboardMock(this.$input)
+                    .focus()
+                    .type(text);
+
+                this.$sendButton.trigger('dxclick');
+            });
+        });
+    });
+
+    QUnit.module('renderMessage', moduleConfig, () => {
+        QUnit.test('Chat items should be updated after renderMessage has been called', function(assert) {
+            const author = {
+                id: MOCK_CURRENT_USER_ID,
+            };
+
+            const newMessage = {
+                author,
+                timestamp: NOW,
+                text: 'NEW MESSAGE',
+            };
+
+            this.instance.renderMessage(newMessage);
+
+            const { items } = this.instance.option();
+            const lastItem = items[items.length - 1];
+
+            assert.strictEqual(lastItem, newMessage);
+        });
+
+        QUnit.test('Message List items should be updated after renderMessage has been called', function(assert) {
+            const author = {
+                id: MOCK_CURRENT_USER_ID,
+            };
+
+            const newMessage = {
+                author,
+                timestamp: NOW,
+                text: 'NEW MESSAGE',
+            };
+
+            this.instance.renderMessage(newMessage);
+
+            const { items } = this.instance._messageList.option();
```

Comment:
missed private API

Reply:
I will consider this test in the next PR